### PR TITLE
Add speaker diarization

### PR DESCRIPTION
**Pull request by alexchirayath**
*Issue #, if available:*

*Description of changes:*

Changes include :
- Transcription API change to include conversational audio
- Updated keys for some components to avoid ReactJS warnings
- Added speaker labels for transcription page(which is propagated on summary page) and left justified text on transcription page

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: app/src/components/AnalysisPane.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ function ResultTable({
           {filteredResults.map((r, i) => (
             <ResultRow
               result={r}
-              key={r.id}
```
Comment (kazbaig) [inline app/src/components/AnalysisPane.js:118]: Why was this done?

File: app/src/components/AnalysisPane.js [hunk 2]
```diff
@@ -139,8 +139,8 @@ export default function AnalysisPane({
 
   return (
     <div className={cs(s.base, visible && s.visible)}>
-      { CATEGORIES.map(cat => (
-        <ResultTable results={allResults} category={cat} onToggleItem={onToggleItem} excludedItems={excludedItems} />
+      { CATEGORIES.map((cat,i)  => (
```
Comment (kazbaig) [inline app/src/components/AnalysisPane.js]: The category should be unique so we can use `cat` as the key

File: app/src/components/SampleSelector.js [hunk 3]
```diff
@@ -46,17 +46,16 @@ export default function SampleSelector({
...
   }, [ onSelect ]);
-
   return (
     <div className={cs(s.base, hidden && s.hidden)}>
       <button className={cs(s.dictateAudio)} onClick={startMic}>Dictate Audio</button>
       <FilePicker onSubmit={filesSubmitted} />
       <div className={s.presets}>
         {Object.keys(samples).map((s, i) => (
-          <SampleButton idx={i + 1} sample={samples[s]} active={activeSample === samples[s]} onSelect={onSelect} onStop={onStop} />
+          <SampleButton idx={i + 1} key={i} sample={samples[s]} active={activeSample === samples[s]} onSelect={onSelect} onStop={onStop} />
```
Comment (kazbaig) [inline app/src/components/SampleSelector.js]: `s` is a unique string here that can be used for the key

File: app/src/components/SampleSelector.js [hunk 4]
```diff
@@ -46,17 +46,16 @@ export default function SampleSelector({
...
         ))}
       </div>
-      <div class={s.credit}>
+      <div className={s.credit}>
```
Comment (kazbaig) [inline app/src/components/SampleSelector.js:53]: 💯

File: app/src/home.js [hunk 5]
```diff
@@ -177,7 +177,23 @@ export default function Home() {
   const history = useHistory();
 
   const addTranscriptChunk = useCallback(({ Alternatives, IsPartial, StartTime }) => {
-    const text = Alternatives[0].Transcript;
+    let addSpeakerLabel = true;
```
Comment (kazbaig) [inline app/src/home.js]: This is really a fold. Wondering if we can change to something like this for readability
```js
const [text] = Alternatives[0].Items.reduce(
  ([prevText, prevAddSpeakerLabel], item) => {
    const isSpeakerChange = item.Type === 'speaker-change';
    const shouldAddSpeakerLabel = !isSpeakerChange && prevAddSpeakerLabel && 'Speaker' in item;
    const isPronunciation = item.Type === 'pronunciation';
    const isPunctuation = item.Type === 'punctuation';

    const speakerLabel = shouldAddSpeakerLabel
      ? `
Speaker ${(parseInt(item.Speaker) + 1).toString()}
`
      : '';

    const itemContent = isPronunciation || isPunctuation ? item.Content : '';

    const spaceAtEnd = isPronunciation ? ' ' : '';

    const text = `${prevText}${speakerLabel}${itemContent}${spaceAtEnd}`;

    const addSpeakerLabel = isSpeakerChange || (shouldAddSpeakerLabel ? false : prevAddSpeakerLabel);

    return [text, addSpeakerLabel];
  },
  ['', true],
);
```

**Review by kazbaig — CHANGES_REQUESTED** [review]
Most comments are about React keys here. [Indexes should be a last resort](https://reactjs.org/docs/lists-and-keys.html#keys).

File: app/src/components/AnalysisPane.js (hunk 1 above)
Comment (alexchirayath) [reply app/src/components/AnalysisPane.js:118]: I think that was a miss. Updated

File: app/src/components/AnalysisPane.js (hunk 2 above)
Comment (alexchirayath) [reply app/src/components/AnalysisPane.js]: Yep..read through the link you shared. Updating across your comments

File: app/src/home.js (hunk 5 above)
Comment (alexchirayath) [reply app/src/home.js]: Yep this is much cleaner. Updated

**Review by kazbaig — APPROVED** [review]

**Pull request merged**